packages/server/graphql/queries/repoIntegrations.ts

```diff
@@ -0,0 +1,51 @@
+import {GraphQLNonNull, GraphQLResolveInfo} from 'graphql'
+import {getUserId} from '../../utils/authorization'
+import standardError from '../../utils/standardError'
+import {GQLContext} from '../graphql'
+import RepoIntegrationQueryPayload from '../types/RepoIntegrationQueryPayload'
+import fetchAllIntegrations, {Integration} from './helpers/fetchAllIntegrations'
+import {getPermsByTaskService, getUserIntegrationIds} from './helpers/repoIntegrationHelpers'
+
+export default {
+  description: 'The integrations that the user would probably like to use',
+  type: new GraphQLNonNull(RepoIntegrationQueryPayload),
+  resolve: async (
+    {teamId, userId}: {teamId: string; userId: string},
+    {first = 10}: {first?: number},
+    context: GQLContext,
+    info: GraphQLResolveInfo
+  ) => {
+    const {authToken, dataLoader} = context
+    const viewerId = getUserId(authToken)
+
+    // AUTH
+    if (userId !== viewerId) {
+      const user = await dataLoader.get('users').load(userId)
+      const {tms} = user!
+      const onTeam = authToken.tms.find((teamId) => tms!.includes(teamId))
+      if (!onTeam) {
+        return standardError(new Error('Not on same team as user'), {userId: viewerId})
+      }
+    }
+    const permLookup = await getPermsByTaskService(dataLoader, teamId, userId)
+    const [userIntegrationIds, allIntegrations] = await Promise.all([
+      getUserIntegrationIds(userId, teamId, permLookup),
+      fetchAllIntegrations(dataLoader, teamId, userId, context, info)
```

Comment:
+1 the primary use case is someone clicking a repo that they have clicked before. For example:
- parabolinc/parabol
- parabolinc/action-devops

If we fetch every single repo from every single integration, we're going to make the user wait a few extra seconds & not provide much value for the primary use case. 

No need to fix now, I wanna get this PR merged,  we'll address this in #6007